extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -130,5 +140,39 @@ void build(CombinedIndexBuildItem index,
                 }
             }
         }
+
+        // manage @BsonProperty for the @ProjectionFor annotation
+        Map<String, String> propertyMapping = new HashMap<>();
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_PROJECTION_FOR)) {
+            Type targetClass = annotationInstance.value().asClass();
+            ClassInfo target = index.getIndex().getClassByName(targetClass.name());
+            for (FieldInfo fieldInfo : target.fields()) {
```

Comment:
Aren't we already collecting all model info? Why do you have to rescan fields and methods? Won't this miss superclass properties?

Reply:
Yes, the model info is available inside the `PanacheMongoRepositoryEnhancer` or the `PanacheMongoEntityEnhancer` but here we only get the fields annotated by `@BsonProperty` from the class denoted by the `@ProjectionFor` annotation. 
This is simpler to do it this way than searching on the two enhancers, then filter the fields.

It uses the same method from Jandex to gather the fields than the enhancer uses, but yes I think it will miss the fields from the super classes if any.